lib/Builtins.cpp

```diff
@@ -14,171 +14,339 @@
 
 #include "Builtins.h"
 
+#include <unordered_map>
+#include <stdlib.h>
+
 using namespace llvm;
+using namespace clspv;
+
+////////////////////////////////////////////////////////////////////////////////
+// utilities for const char* map keys
+struct cstr_hash {
+  std::size_t operator()(const char *cstr) const {
+    std::size_t hash = 5381;
```

Comment:
Where did you get this hash function?  Why this hash function?

Reply:
Random online.. I will switch to FNV-1a.